### PR TITLE
Upgrade mypy to 1.8.0

### PR DESCRIPTION
**Pull request by hussein-awala**

### CODE REVIEW

**Review by dirrao — CHANGES_REQUESTED** [review]
I see some pre commit checks are failing due to version upgrade. May be the new version enforces the strict checks.

File: airflow/providers/http/hooks/http.py [hunk 1]
```diff
@@ -242,9 +243,11 @@ def run_with_advanced_retry(self, _retry_args: dict[Any, Any], *args: Any, **kwa
...
 
         """
-        self._retry_obj = tenacity.Retrying(**_retry_args)
-
-        return self._retry_obj(self.run, *args, **kwargs)
+        # Work around "Never not callable" error in Mypy 1.8.
+        # This is probably an annotation issue in tenacity? Not entirely sure.
+        retrying = tenacity.Retrying(**_retry_args)
+        self._retry_obj = retrying
+        return retrying(self.run, *args, **kwargs)
```
Comment (hussein-awala) [inline airflow/providers/http/hooks/http.py]: Another alternative solution:
```suggestion
        self._retry_obj = cast(Callable, tenacity.Retrying(**_retry_args))
        return self._retry_obj(self.run, *args, **kwargs)
```

Comment (potiuk) [reply airflow/providers/http/hooks/http.py]: Maybe a good idea to open an issue to tenacity ? Last time when I did it, they fixed it really quickly https://github.com/jd/tenacity/issues/389

Comment (hussein-awala) [reply airflow/providers/http/hooks/http.py]: I just created https://github.com/jd/tenacity/issues/428

File: airflow/providers/http/hooks/http.py [hunk 2]
```diff
@@ -258,7 +258,8 @@ def run_with_advanced_retry(self, _retry_args: dict[Any, Any], *args: Any, **kwa
         """
         self._retry_obj = tenacity.Retrying(**_retry_args)
 
-        return self._retry_obj(self.run, *args, **kwargs)
+        # TODO: remove ignore type when https://github.com/jd/tenacity/issues/428 is resolved
+        return self._retry_obj(self.run, *args, **kwargs)  # type: ignore
```
Comment (hussein-awala) [inline airflow/providers/http/hooks/http.py:262]: I propose merging this PR with this comment, and once the issue is resolved, we can remove the ignore comment. WDYT?

Comment (potiuk) [reply airflow/providers/http/hooks/http.py:262]: Yes

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Seems it works.

**Pull request merged**